plugin/pkg/auth/authenticator/token/oidc/oidc.go

```diff
@@ -95,69 +126,85 @@ func New(opts OIDCOptions) (*OIDCAuthenticator, error) {
 		TLSClientConfig: &tls.Config{RootCAs: roots},
 	})
 
-	hc := &http.Client{}
-	hc.Transport = tr
+	return &OIDCAuthenticator{
+		issuerURL:       opts.IssuerURL,
+		trustedClientID: opts.ClientID,
+		usernameClaim:   opts.UsernameClaim,
+		groupsClaim:     opts.GroupsClaim,
+		httpClient:      &http.Client{Transport: tr},
+	}, nil
+}
 
-	maxRetries := opts.MaxRetries
-	if maxRetries < 0 {
-		maxRetries = DefaultRetries
-	}
-	retryBackoff := opts.RetryBackoff
-	if retryBackoff < 0 {
-		retryBackoff = DefaultBackoff
+// Close closes the OIDC authenticator, this will close the provider sync goroutine.
+func (a *OIDCAuthenticator) Close() {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	a.closed = true
+
+	if a.close != nil {
+		a.close()
 	}
+	return
+}
 
-	for i := 0; i <= maxRetries; i++ {
-		if i == maxRetries {
-			return nil, fmt.Errorf("failed to fetch provider config after %v retries", maxRetries)
-		}
+func (a *OIDCAuthenticator) client() (*oidc.Client, error) {
```

Comment:
since oidcClient is typically going to be written exactly once, then read forever, it's probably worth using an `atomic.Value` and doing a `Load()` check to see if it has a non-nil value before making all incoming requests acquire a lock. a quick [benchmark](https://gist.github.com/liggitt/455adb9f7627ec4fe0439f8e14ea456b) shows that to be about 100x faster:

```
  oidcClient atomic.Value // Contains a *oidc.Client. Do not access directly. Use client() method.
}
...
func (a *OIDCAuthenticator) client() (*oidc.Client, error) {
  // Fast check to see if it's already initialized
  if client := a.oidcClient.Load(); client != nil {
    return client.(*oidc.Client), nil
  }

  // Acquire lock, then recheck if it's initialized
  a.mu.Lock()
  defer a.mu.Unlock()
  if client := a.oidcClient.Load(); client != nil {
    return client.(*oidc.Client), nil
  }

  // Initialize
  client := ... init client ...
  a.oidcClient.Store(client)
  return client, nil
}
```